types/pub_rand_commit.go

```diff
@@ -1,35 +1,17 @@
 package types
 
 import (
-	"fmt"
-
 	bbn "github.com/babylonlabs-io/babylon/v3/types"
 	"github.com/btcsuite/btcd/btcec/v2"
 	"github.com/cometbft/cometbft/crypto/merkle"
 )
 
-// TODO: Shared across Babylon/Cosmos/Rollup FPs. Sources return different field
-// names (epoch_num/height/babylon_epoch). Consider source-specific structs or
-// interface approach.
-type PubRandCommit struct {
-	StartHeight uint64 `json:"start_height"`
-	NumPubRand  uint64 `json:"num_pub_rand"`
-	Commitment  []byte `json:"commitment"`
-}
-
-// Validate checks if the PubRandCommit structure is valid
-// returns an error if not.
-func (prc *PubRandCommit) Validate() error {
-	if prc.NumPubRand < 1 {
-		return fmt.Errorf("NumPubRand must be >= 1, got %d", prc.NumPubRand)
-	}
-
-	return nil
-}
-
-// EndHeight returns the last height covered by this public randomness commitment
-func (prc *PubRandCommit) EndHeight() uint64 {
-	return prc.StartHeight + prc.NumPubRand - 1
+// PubRandCommit interface abstracts source-specific structs
+// (Babylon/Cosmos/Rollup FPs) to handle different JSON field names across
+// sources.
+type PubRandCommit interface {
+	EndHeight() uint64
+	Validate() error
 }
 
```

Comment:
We need to define getter functions of common fields in the interface, including StartHeight, NumPubRand, and Commitment.